feature: Add Docker build (#55)
anitab-org/open-source-programs-backend: README.md

5. To run the migrations run: `python manage.py migrate`
6. You can terminate the process by `Ctrl+C` in your terminal.

> ### Run with Docker

Reviewer: ```suggestion
### Run with Docker

```

> 1. Make sure latest version of docker-compose is installed by running ``docker-compose -v``. If it is not,install using ``pip install -U docker-compose``

Reviewer: ```suggestion
1. Make sure the latest version of docker-compose is installed by running ``docker-compose -v``. If it is not, install using ``pip install -U docker-compose``
```

2. Change Database HOST in ``settings.py``

- Go to ``main/settings.py``
- Under databases change host to db.

3. **Build docker image**

Execute ``sudo docker-compose run web`` to build the image.

4. **Run docker image**

> Run ``sudo docker-compose up`` from the root directory of project.

Reviewer: as step 4 you could have an additional step with the link where the server is running, so we don't have to look at dockerfiles to test it live. is it localhost:8000?

Author: Yes it is localhost:8000

5. To run the migrations run: `python manage.py migrate`
6. You can terminate the process by `Ctrl+C` in your terminal.

### Run with Docker

> 1. Make sure the latest version of docker-compose is installed by running ``docker-compose -v``. If it is not, install using ``pip install -U docker-compose``

Reviewer 2: @devkapilbansal just add `sudo apt install docker.io` for installing docker before the 1st step in the README for **Run with Docker**. Also u can include the version check `docker -v`  as u have mentioned in the step above.

2. Change Database HOST in ``settings.py``

- Go to ``main/settings.py``
- Under databases change host to db.

3. **Build docker image**

> Execute ``sudo docker-compose run web`` to build the image.

Reviewer 3: @devkapilbansal This command is not for building the docker image.
```suggestion
Execute ``sudo docker-compose up`` to build the image.
```

4. **Run docker image**

> Run ``sudo docker-compose up`` from the root directory of project. Navigate to `http://localhost:8000` in your browser.

Reviewer 3: @devkapilbansal This should be used for running server.
```suggestion
Run ``sudo docker-compose run websudo ss -lptn 'sport = :5432'`` from the root directory of project. Navigate to `http://localhost:8000` in your browser.
```

> 

Reviewer 3: There should be a command for migrating in docker. Just a suggestion :)
```suggestion

5. **Migrating Database**

To run the migrations run ``sudo docker-compose run web python manage.py migrate``.
```

Author: @codesankalp the user can log into the container once build and run any command he wants.

`docker exec -it {container_name} /bin/bash` will open a terminal for the same.

    python manage.py runserver
    ```
8. Navigate to `http://localhost:8000/` in your browser.

> 9. To change the port you may run `python manage.py runserver 0.0.0.0:<port_number>`

Author: Removed `0.0.0.0` as it is not required

To run the tests run: `python manage.py test`.

### Run with Docker

1. Make sure the latest version of docker and docker-compose are installed.

- For **docker** run ``docker -v`` from your terminal. If docker is not installed, install it from [here](https://docs.docker.com/engine/install/).
- For **docker-compose** run ``docker-compose -v`` from terminal. If it is not installed then install it by running ``pip install docker-compose`` or from [here](https://docs.docker.com/compose/install/).

2. Change Database HOST in ``settings.py``

- Go to ``main/settings.py``
- Under databases change host to db.

3. **Build docker image**

Execute ``sudo docker-compose run web`` to build the image.

4. **Run docker image**

Run ``sudo docker-compose up`` from the root directory of project. Navigate to `http://localhost:8000` in your browser.

**Note** 
- Run `docker-compose up --build` to rebuild the image.

> - To **interact** with docker containers use ``docker exec -it {container id} bash`` from your terminal. Container id can be found using ``docker ps``.

Author: @codesankalp Added this command if someone wants to run commands like `python manage.py migrate`.

Reviewer 3: Thanks @devkapilbansal , for adding this.
